controllers/devopsconfig/yaml_merge_test.go

```diff
@@ -0,0 +1,132 @@
+package devopsconfig
+
+import (
+	jsonpatch "github.com/evanphx/json-patch/v5"
+	"github.com/imdario/mergo"
+	"gotest.tools/assert"
+	"k8s.io/apimachinery/pkg/util/json"
+	"log"
+	"sigs.k8s.io/yaml"
+	"testing"
+)
+
+func TestYamlMerge(t *testing.T) {
+	var data = `
+a: Easy!
+b:
+ c: 2
+ d: [3, 4]
+e: true
+f:
+ g: true
+`
+
+	var patch = `
+a: Not Easy!
+b:
+ h: new field
+e: false
+f:
+ g: false
+`
+	dataJson, err := yaml.YAMLToJSON([]byte(data))
+
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	log.Printf("data json: %s", dataJson)
+
+	patchJson, err := yaml.YAMLToJSON([]byte(patch))
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	log.Printf("patch json: %s", patchJson)
+
+	mergedJson, err := jsonpatch.MergePatch(dataJson, patchJson)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	log.Printf("merged json: %s", mergedJson)
+
+	dataMap := make(map[string]interface{})
+	err = json.Unmarshal(mergedJson, &dataMap)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	assert.Equal(t, "Not Easy!", dataMap["a"])
+	assert.Equal(t, "new field", dataMap["b"].(map[string]interface{})["h"])
+	assert.Equal(t, int64(2), dataMap["b"].(map[string]interface{})["c"])
+	assert.Equal(t, false, dataMap["e"])
+	assert.Equal(t, false, dataMap["f"].(map[string]interface{})["g"])
+}
+
+func TestArrayYamlMerge(t *testing.T) {
+	var data = `
+data:
+  - name: 'hello'
+    value: 'world'
+  - name: 'name1'
+    value: 'value1'
+status:
+  enabled: true
+`
+
+	var patch = `
+data:
+  - name: 'hello'
+    value: 'new world'
+`
+
+	var dataMap = make(map[string]interface{})
+	err := yaml.Unmarshal([]byte(data), &dataMap)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	log.Printf("data map: %v", dataMap)
+
+	var patchMap = make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(patch), &patchMap)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	log.Printf("patch map: %v", patchMap)
+
+	err = mergo.Merge(&dataMap, patchMap, mergo.WithAppendSlice)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	log.Printf("merged map: %v", dataMap)
+}
+
+func TestFindItem(t *testing.T) {
+	mapSlice := []map[string]interface{}{
+		{
+			"key1": "value1",
+		},
+		{
+			"key2": "value2",
+		},
+		{
+			"key3": "value3",
+		},
+	}
+
+	item := findItem(mapSlice, "key1", "value1")
+	assert.Assert(t, item != nil)
+
+	item = findItem(mapSlice, "key2", "value1")
+	assert.Assert(t, item == nil)
+
+	item = findItem(mapSlice, "key100", "value1")
+	assert.Assert(t, item == nil)
```

Comment:
The coverage of the new code lines is too low.

![image](https://user-images.githubusercontent.com/1450685/124411214-b196cd00-dd7e-11eb-8fb3-9d99181be02c.png)